### DISCUSSION TITLE
Indexed properties as environment variables?

### DISCUSSION THREAD

**Discussion by edeandrea** [quarkusio/quarkus#26446] [Q&A] [answered]
Looking at https://quarkus.io/guides/config-reference#indexed-properties, how would I express the following

```properties
my.indexed.collection[0].name=dog
my.indexed.collection[1].name=cat
my.indexed.collection[2].name=turtle
```

As environment variables?

```
MY_INDEXED_COLLECTION_0_NAME=dog
MY_INDEXED_COLLECTION_1_NAME=cat
MY_INDEXED_COLLECTION_2_NAME=turtle
```

Does not seem to work.

**Comment by gsmet** [comment, marked as answer]
Have you tried something like:
```
MY_INDEXED_COLLECTION_0__NAME=dog
MY_INDEXED_COLLECTION_1__NAME=cat
MY_INDEXED_COLLECTION_2__NAME=turtle
```
IIRC special characters are replaced with a _. But it might not work for indexes.

**Reply by edeandrea** [reply]
That works gsmet. Thanks!

Should this be documented somewhere? If it is, I didn't find it.

**Reply by gsmet** [reply]
If it's not and you can find a good place to add it, please do. We had a few questions about that already.

**Reply by edeandrea** [reply]
I would think it should first be documented in the upstream and then carried over to the Quarkus documentation?

It seems the examples and the docs we have here on our site are mostly the same as from https://smallrye.io/docs/smallrye-config/config/config.html yet I don't see anything there related to the rules on reading from environment variables.

It's also very lightly documented on https://download.eclipse.org/microprofile/microprofile-config-3.0/microprofile-config-spec-3.0.html#default_configsources.env.mapping

**Reply by radcortez** [reply]
For environment variables in the Quarkus docs we refer to upstreams:
https://quarkus.io/guides/config-reference#environment-variables

Which link here:
https://smallrye.io/smallrye-config/latest/config/environment-variables/

We can of course improve all the documentation. We recently rewrote the SmallRye Config docs, but the main webpage is still linking to the old docs (Quarkus docs already link to the right one).

@edeandrea what do you think?

**Reply by edeandrea** [reply]
Thanks @radcortez.

I think the docs look good. I think what was misleading was

> replace each character that is neither alphanumeric nor `_` with `_`

which makes sense that `MY_INDEXED_COLLECTION_0__NAME=dog` works, since there is a `]` followed by a `.`, which when both are replaced with `_`, you get two `__` in a row.

Just a Jedi mind trick :)

**Reply by radcortez** [reply]
The format looks a bit weird, but I ended up going with that format to keep it consistent with everything else.

**Reply by edeandrea** [reply]
I don't think there's anything else we need to do here.

**Comment by edeandrea** [comment]
I tried

```
MY_INDEXED_COLLECTION__0__NAME=dog
MY_INDEXED_COLLECTION__1__NAME=cat
MY_INDEXED_COLLECTION__2__NAME=turtle
```

and that did not work. I will try what you mentioned.
